slither/detectors/naming_convention/naming_convention.py

```diff
@@ -167,10 +175,22 @@ def _detect(self) -> List[Output]:
                         results.append(res)
 
                 else:
-                    if var.visibility in ["private", "internal"]:
+                    # first priority: check for immutable variable naming conventions
+                    if var.is_immutable:
+                        correct_naming = self.is_immutable_naming(var.name)
+
+                    # second priority: check for state variable naming conventions
+                    elif self.is_state_naming(var.name):
+                        correct_naming = True
```

Comment:
I think these should also be guarded by a check that ensures they're private/immutable as public variables should still be camel case. If this convention is being used, I think public getters with camel cases are created for internal/private variables with the i_/s_ prefix